### DISCUSSION TITLE
User stories: please share how your organization is adopting Lima

### DISCUSSION THREAD

**Discussion by AkihiroSuda** [lima-vm/lima#2390] [Show and tell]
We, the maintainers of Lima, are trying to promote Lima from CNCF Sandbox to CNCF Incubating:
- https://github.com/cncf/toc/issues/1348

To help the due diligence process at CNCF, we would like to collect user stories 🙏.

Please post:

- Name of your organization
- Host OS
- Guest OS (distro)
- Application inside the guest OS
- Why you adopted Lima
- Comments

**Comment by rfay** [comment]
DDEV supports Lima directly, in Rancher Desktop, and as part of Colima.

* DDEV Foundation https://ddev.com, https://ddev.readthedocs.io/en/stable/users/install/docker-installation/#macos
* Host OS: macOS (amd64 and arm64)
* Guest OS: Linux
* Application in guest: nginx/php-fpm/etc.
* Why we adopted: Open-source, well-maintained, great solution

Current voluntary telemetry for DDEV shows about 10,000 weekly users. DDEV is the preferred/specified/documented local dev recommendation for TYPO3, Drupal, Craft CMS, and a few other PHP CMSs.

**Comment by bloodeagle40234** [comment]
- NTT Communications Corporation https://www.ntt.com/en/index.html
- Host OS: macOS (arm64)
- Guest OS (distro): Linux (manily Ubuntu)
- Application inside the guest OS: nginx, nerdctl, lustre all in one (for testing)
- Why you adopted Lima: well-maintained, easy-to-install/use, creating Linux env in my laptop

**Comment by deviantintegral** [comment]
- Name of your organization: Lullabot
- Host OS: macOS
- Guest OS (distro): Ubuntu (really, whatever the default docker template uses)
- Application inside the guest OS: ddev, but also anything else on Docker.
- Why you adopted Lima: Performance, simplicity, open-source, responsive to end user needs.
- Comments: Our team can choose their development environments. Most use macOS, though some use Linux and we have 1-2 Windows users. Docker Desktop was very buggy for years, and prioritized features we didn't care about instead of core stability and performance. Their difficult licensing changes made Lima and colima a clear winner. While some team members do choose to pay for and use Docker Desktop, most are happy with Lima.

**Comment by Shubhranshu153** [comment]
- Name of your organization:  Finch, Lima is used in Finch  (https://runfinch.com/)
- Host OS: MacOS, Windows
- Guest OS (distro): Fedora
- Application inside the guest OS:we began looking at the available open source components for container tooling and were immediately impressed with the progress of Lima, recently included in the Cloud Native Computing Foundation (CNCF) as a sandbox project. [https://aws.amazon.com/blogs/opensource/introducing-finch-an-open-source-client-for-container-development/](url)
- Comments: Finch has 40k downloads, 54k views, 12k clones

**Comment by nirs** [comment]
- Name of your organization: RamenDr (https://github.com/RamenDR)
- Host OS: macOS
- Guest OS (distro): Ubuntu
- Application inside the guest OS: Complex Kubernetes cluster (based on Lima k8s template) running OCM, Rook Ceph storage system, ArgoCD, Submariner, VolSync, MinIO, and RamenDR. We run 3 clusters configured to replicate storage between clusters, and run full disaster recovery flow for testing RamenDR. 
- Why you adopted Lima: The killer feature for us is Rosetta support, since we don't have arm64 images for all the dependencies. Another critical feature is socket_vmnet support, allowing access between vms and from vm to host. Lima is also easy install, great for automation, very flexible and fully featrured, has active and friendly community, and is also fun project to contribute to.
- Comments: The same test infrastructure run on Linux using Minikube clusters. MacBooks are very popular for developers and with Lima we can now run complicated end to end tests locally on the preferred hardware.

**Comment by hblixt** [comment]
**Name of your organization:** [Intuit](https://www.intuit.com/)
**Host OS**: 95% MacOS/5% Windows
**Guest OS (distro)**: Alpine (through Rancher Desktop)
**Application inside the guest OS:** Currently, we only use it as part of a ~7000 seat Rancher Desktop rollout. 
**Why you adopted Lima:** As mentioned above, it was through rolling out Rancher Desktop through our organization.  In our testing and usage, it has fulfilled all the requirements we had and we've had no issues or concerns with it. 
**Comments:** Working with the Lima and Rancher community has been great. Intuit uses a lot of open source and CNCF projects and this has really stood out as great project and community to use and work with.

**Comment by BenTheElder** [comment]
- Name of your Organization: The Kubernetes Project, [KIND](https://github.com/kubernetes-sigs/kind) subproject
- Host OS: ~macOS GitHub Actions Runners~ Ubuntu 24.04 via GitHub Actions Runners
- Guest OS (distro): Fedora
- Application inside the guest OS: Docker, podman, nerdctl/containerd and KIND
- Why you adopted Lima: We wanted a permissively licensed solution for VMs for testing various kernel environments.
- Comments: I'm also looking into a guide for using Lima to develop the core Kubernetes project, I and other contributors have been using it for some time, with a few configuration tweaks it works really well and allows developing with `kind` on macOS. Lima is a valuable tool for the ecosystem.

**Reply by rfay** [reply]
I assume you're using amd64 macOS runners? arm64 don't work do they, because of Apple's problem with nested virtualization?

**Reply by BenTheElder** [reply]
Oh, upon digging back into that, I forgot that we've moved from the macOS runners with the nested virt hack to the ubuntu runners.

One of the current workflows is here, on ubuntu-24.04 https://github.com/kubernetes-sigs/kind/blob/ea4506ec0f98511e7e7f404b7102fb522287fa3b/.github/workflows/vm.yaml

Updating my comment.

**Reply by jandubois** [reply]
The problem is that the GitHub runners are using M1/M2 CPUs. Nested virtualization requires M3+. It is a hardware issue, not a software one.

**Reply by rfay** [reply]
Actually... I can't do nested virtualization on M4. I think they just refuse it. But you know more than I do. Parallels can't do it, etc.

**Reply by jandubois** [reply]
This is a limitation of Parallels because they didn't implement it. It works with Lima and UTM.

https://developer.apple.com/documentation/virtualization/vzgenericplatformconfiguration/isnestedvirtualizationsupported

**Reply by rfay** [reply]
They certainly claim otherwise :) And other virtualization environments also fail the same. Interesting that it works with Lima and UTM
